inst/templates/ad_advs.R

```diff
@@ -0,0 +1,277 @@
+advs <- advs %>%
+  mutate(
+    ATPT = VSTPT,
+    ATPTN = VSTPTNUM,
+    AVISIT = case_when(
+      is.na(VISIT) ~ NA_character_,
+      VSTESTCD == "HEIGHT" & VISIT == "SCREENING 1" ~ "Screening",
+      str_detect(VISIT, "SCREEN|UNSCHED|RETRIEVAL|AMBUL") ~ NA_character_,
+      TRUE ~ str_to_title(VISIT)
+    ),
+    AVISITN = case_when(
+      VISIT == "BASELINE" ~ 0,
+      VSTESTCD == "HEIGHT" & VISIT == "SCREENING 1" ~ -1,
+      str_detect(VISIT, "WEEK") ~ as.integer(str_extract(VISIT, "\\d+")),
+      TRUE ~ NA_integer_
+    )
+  )
+
+
+# Derive results ----
+# See the "Derive Results (AVAL, AVALC)" vignette section for more information:
+# (https://pharmaverse.github.io/admiral/articles/bds_finding.html#aval)
+
+# Derive analysis result (AVAL)
+advs <- advs %>%
+  mutate(AVAL = VSSTRESN)
+
+
+# Derive domain specific variables ----
+# See the "Derive Additional Parameters" vignette section for more information:
+# (https://pharmaverse.github.io/admiral/articles/bds_finding.html#derive_param)
+
+# Derive BMI
+advs <- advs %>%
+  filter(VSTESTCD != "BMI") %>%
+  derive_param_bmi(
+    by_vars = exprs(
+      !!!get_admiral_option("subject_keys"), !!!adsl_vars,
+      AVISIT, AVISITN, ADT, ADY, ATPT, ATPTN
+    ),
+    set_values_to = exprs(
+      PARAMCD = "BMI"
+    ),
+    get_unit_expr = VSSTRESU,
+    constant_by_vars = get_admiral_option("subject_keys")
+  )
+
+
+# Derive categorization variables ----
+# See the "Derive Categorization Variables" vignette section for more
+# information:
+# (https://pharmaverse.github.io/admiral/articles/bds_finding.html#cat)
+
+# Create analysis categories look-up (conditional look-up) table
+avalcat_lookup <- exprs(
+  ~PARAMCD, ~condition,                ~AVALCAT1,           ~AVALCA1N,
+  "BMI",    AVAL < 18.5,               "Underweight",       1,
+  "BMI",    AVAL >= 18.5 & AVAL < 25,  "Normal weight",     2,
+  "BMI",    AVAL >= 25 & AVAL < 30,    "Overweight",        3,
+  "BMI",    AVAL >= 30 & AVAL < 35,    "Obesity class I",   4,
+  "BMI",    AVAL >= 35 & AVAL < 40,    "Obesity class II",  5,
+  "BMI",    AVAL >= 40,                "Obesity class III", 6,
+  "BMI",    is.na(AVAL),               NA_character_,       NA_integer_
+)
+
+# Derive BMI class (AVALCAT1, AVALCA1N)
+advs <- advs %>%
+  derive_vars_cat(
+    definition = avalcat_lookup,
+    by_vars = exprs(PARAMCD)
+  )
+
+
+# Derive Baseline variables ----
+# See the "Derive Baseline" and "Derive Change from Baseline " vignette sections
+# for more information:
+# (https://pharmaverse.github.io/admiral/articles/bds_finding.html#baseline)
+# (https://pharmaverse.github.io/admiral/articles/bds_finding.html#bchange)
+
+# Add baseline flag (ABLFL)
+advs <- advs %>%
+  restrict_derivation(
+    derivation = derive_var_extreme_flag,
+    args = params(
+      by_vars = exprs(!!!get_admiral_option("subject_keys"), PARAMCD),
+      order = exprs(ADT, ATPTN, AVISITN),
```

Comment:
This dataset is a bit strange when it comes to blood pressure. In reality, what one usually does in blood pressure is that you measure the blood pressure a minimum of three times, and report the average of the last two measurements. It is from my perspective unclear what we actually have in the test data.

However, this strange grouping actually happened after we changed from using "VISITNUM" to "AVISITN" in this function.

I have just quickly looked at the admiral BDS template, where it appears that they actually have many baseline values and have change for each basetype. I am changing back to VISITNUM, and then we can later discuss if this might be confusing.

Reply:
I spoke to soon. It appears that this is not the case. The different ATPT's are actually used as basetypes in the admiral vignette (https://pharmaverse.github.io/admiral/articles/bds_finding.html#baseline). 

I don't think adding basetypes to our vignette would make to much sense, and would argue that the way it is (where we use the last measurements at baseline) might be best for now. I do still agree that it is messy, but cannot think of anything that makes perfect sense right now.

Reply:
Ok, let's keep that in mind and potentially change it in one of the next PRs.

Reply:
```suggestion
    derivation = derive_var_extreme_flag,
    args = params(
      by_vars = exprs(!!!get_admiral_option("subject_keys"), PARAMCD, ATPTN),
      order = exprs(ADT, ATPTN, AVISITN),
```
IMO we should flag with ABLFL=Y not only records with ATPT = "AFTER STANDING FOR 3 MINUTES". 
Why not add ATPTN to the by_vars function argument?

I also agree with @AndersAskeland that the average of SYSBP/DIABP measurements is often taken into account. This means that another parameter should be created. Please decide if it would be beneficial to the user.